### DISCUSSION TITLE
Optimize how MQClientAPIExt is obtained in the MQClientAPIFactory

### DISCUSSION THREAD

**Discussion by socutes** [apache/rocketmq#6330] [Ideas]
To discuss whether this problem needs to be optimized:
The getClient method in the current MQClientAPIFactory. The MQClientAPIExt is obtained by random seeds.In extreme cases, load imbalance on MQClientAPIExt may occur.
You are advised to change the obtaining mode from random to polling.
![image](https://user-images.githubusercontent.com/7479320/224696328-4a8073db-6609-40e9-a726-493daa5e4904.png)

https://github.com/apache/rocketmq/issues/6312

**Comment by echooymxq** [comment]
用轮询应该需要一个并发计数器了，性能并不一定好.

**Reply by socutes** [reply]
Its advantage is that when there are many clients, the traffic can be guaranteed to be balanced. @xdkxlk What do you think?

**Reply by xdkxlk** [reply]
> Its advantage is that when there are many clients, the traffic can be guaranteed to be balanced. @xdkxlk What do you think?

I think using random should be enough. The `getClient` method will be called frequently, so the distribution overall should be relatively even.
